src/base/static/js/views/app-view.js

```diff
@@ -846,52 +699,100 @@ var Shareabouts = Shareabouts || {};
-      // If the model was found in the places, go ahead and use it.
-      if (model) {
-        onPlaceFound(model);
+      bindCollectionsListeners = function(collections, property, type, finalCollection) {
+        var numCollections = _.keys(collections).length,
+        i = 0,
+        searchTerm = {};
+        searchTerm[property] = args.modelId;
+        _.each(collections, function(collection, datasetId) {
+          collection.on("sync", function(syncedCollection) {
+            i++;
+            var model = syncedCollection.where(searchTerm);
+            if (model.length === 1) {
+              foundInCallback = true;
+              onFound(model[0], type, datasetId);
+            } else if (numCollections === i && finalCollection && !foundInCallback) {
+              // if this is the last collection of the set and the final
+              // set of collections and no model has been found, it means the
+              // model doesn't exist.
+              onNotFound();
+            }
+          });
+        });
+      };
 
-      // Otherwise, fetch and use the result.
+      if (args.datasetSlug) {
+        // If we have a slug, we definitely have a place model
+        var datasetId = _.find(self.options.mapConfig.layers, function(layer) { 
+          return layer.slug === args.datasetSlug;
+        }).id;
+        model = this.places[datasetId].get(args.modelId);
+        if (model) {
+          onFound(model, "place", datasetId);
+          return;
+        } else {
+          this.places[datasetId].fetchById(args.modelId, {
+            validate: true,
+            success: function(model) {
+              onFound(model, "place", datasetId);
+              return;
+            },
+            error: function() {
+              onNotFound();
+              return;
+            },
+            data: {
+              include_submissions: includeSubmissions
+            }
+          });
+        }
       } else {
-        this.places[datasetId].fetchById(modelId, {
-          // Check for a valid location type before adding it to the collection
-          validate: true,
-          success: onPlaceFound,
-          error: onPlaceNotFound,
-          data: {
-            include_submissions: includeSubmissions
-          }
-        });
+        // Otherwise, we have a landmark-style url, which may correspond
+        // to a place or a landmark.
+        // Conduct a search according to the following strategy: 
+        // 1. check loaded place collections
+        // 2. check loaded landmark collections
+        // 3. set up sync listeners on all place and landmark collections
+        if (searchLoadedCollections(this.places, "url-title", "place")) {
+          return;
+        };
+        if (searchLoadedCollections(this.landmarks, "id", "landmark")) {
+          return;
+        };
+        bindCollectionsListeners(this.places, "url-title", "place", false);
+        bindCollectionsListeners(this.landmarks, "id", "landmark", true);
```

Comment:
We may want to consider unbinding these listeners after we have finished loading/searching the collections.